src/index.js

```diff
@@ -0,0 +1,210 @@
+var multiaddr = require('multiaddr')
+var config = require('./config')
+var requestAPI = require('./request-api')
+
+exports = module.exports = IpfsAPI
+
+function IpfsAPI (host_or_multiaddr, port) {
+  var self = this
+
+  if (!(self instanceof IpfsAPI)) {
+    return new IpfsAPI(host_or_multiaddr, port)
+  }
+
+  try {
+    var maddr = multiaddr(host_or_multiaddr).nodeAddress()
+    config.host = maddr.address
+    config.port = maddr.port
+  } catch (e) {
+    config.host = host_or_multiaddr
+    config.port = port || config.port
+  }
+
+  // -- Internal
+
+  function command (name) {
+    return function (cb) {
+      return requestAPI(name, null, null, null, cb)
+    }
+  }
+
+  function argCommand (name) {
+    return function (arg, cb) {
+      return requestAPI(name, arg, null, null, cb)
+    }
+  }
+
+  // -- Interface
+
+  self.send = requestAPI
+
+  self.add = function (files, opts, cb) {
+    if (typeof (opts) === 'function' && cb === undefined) {
+      cb = opts
+      opts = {}
+    }
+
+    return requestAPI('add', null, opts, files, cb)
+  }
```

Comment:
so, there's a bunch of commands that take options but that the api doesnt make this available. it may be better if we make _all_ commands capable of taking opts field, and passing it through to go-ipfs? (may be better as a PR on top of this one though.